### ISSUE TITLE
Ability to determine if a dependency is a development dependency

### ISSUE THREAD

**Issue by DarthHater** [voxpelli/list-installed#23]
Something I really liked about `read-installed` was that you could:

```
2.0.0
Breaking changes in 2.0.0:

The second argument is now an Object that contains the following keys:

depth optional, defaults to Infinity
log optional log Function
dev optional, default false, set to true to include devDependencies
```

The dev flag was super useful, in that you could exclude dev dependencies. This is useful in the tools we craft at Sonatype, because a lot of the times developers may want to weed out some of the noise the dev dependencies cause.

It would be nifty if this was available as an option for getting the list, but if there was a property on the returned objects with `development: t/f` that would be great too!

**Comment by voxpelli** [comment]
Currently this module doesn't read anything about what _should_ be installed, only that _something_ is installed, so it can't tell if it's due to it being a dependency, a dev dependency, a hoisted dependency or something else.

To achieve this one would have to pair the current functionality with a filter that filters the output.

Maybe I could add a `filter` option and one could then do:

```javascript
const { devDependencies = {} } = await readPkg();

const installed = await listInstalled('./', { filter: name => !!devDependencies[name] });
```

That would keep a good separation of concern why still achieving your goal and doing so without reading any non-needed `package.json` files. What do you think @DarthHater?

**Comment by DarthHater** [comment]
I love it! Agnostic to the solution, just the ability to accomplish it.

**Comment by voxpelli** [comment]
Haven't forgot about this, I got preoccupied by #25 and by life 🙈

**Comment by voxpelli** [comment]
@DarthHater @bhamail Sorry for the extreme delay here, shipped this as `4.1.0` now
